numpy/lib/_function_base_impl.py

```diff
@@ -3827,8 +3827,8 @@ def sinc(x):
 
     """
     x = np.asanyarray(x)
-    x = x.astype(np.float64) if np.isdtype(x.dtype, ('bool', 'integral')) else x
-    y = pi * where(x, x, np.finfo(x.dtype).smallest_normal)
+    x *= pi
+    y = where(x, x, np.finfo(x.dtype).eps)
```

Comment:
```suggestion
    if dtype.kind == "f":
        eps = np.finfo(x.dtype).eps
    else:
        # Hope that 1e-20 is sufficient for objects...
        eps = 1e-20
    y = where(x, x, eps)
```
Can you add `longdouble` to the tests (and maybe also object) because those would have failed before, I think.

Reply:
looks like object has never been supported

```python
In [1]: import numpy as np

In [2]: x = np.arange(4, dtype=object)

In [3]: np.sinc(x)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
AttributeError: 'float' object has no attribute 'sin'

The above exception was the direct cause of the following exception:

TypeError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 np.sinc(x)

File ~/micromamba/envs/sklearn-env/lib/python3.13/site-packages/numpy/lib/_function_base_impl.py:3785, in sinc(x)
   3783 x = np.asanyarray(x)
   3784 y = pi * where(x == 0, 1.0e-20, x)
-> 3785 return sin(y)/y

TypeError: loop of ufunc does not support argument 0 of type float which has no callable sin method
```

Reply:
how can we coerce `pi * x` to longdouble without re-introducing `result_type`?

Reply:
Ah, right... so that would need an object that has a `.sin()`.  So maybe we don't need it (or maybe just add without a test)?